cmd/otelcol/main.go

```diff
@@ -109,44 +112,27 @@ func contains(arr []string, str string) bool {
 
 // Get the value of a key in an array
 // Support key/value with and with an equal sign
-func getKeyValue(args []string, argName string) string {
-	val := ""
-	for i, arg := range args {
+func getKeyValue(args []string, arg string) string {
+	argEq := arg + "="
+	for i := range args {
 		switch {
-		case strings.HasPrefix(arg, argName+"="):
-			s := strings.Split(arg, "=")
-			val = s[1]
-		case arg == argName:
-			i++
-			val = args[i]
+		case strings.HasPrefix(args[i], argEq):
+			if s := strings.SplitN(args[i], "=", 2); len(s) == 2 {
```

Comment:
When would `len(s)` not be 2?

Reply:
You've resolved this without answering or removing the seemingly unnecessary `len(s) == 2` condition

Reply:
Sorry, weird! I actually remember giving an example where I think len(s) can be equal to 1 when for instance args[i]="config=".

Reply:
`strings.HasPrefix(args[i], argEq)` would be true in that case though?

Reply:
Yes, I remove the unnecessary check